blocks/snow.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package blocks
+
+import (
+	"time"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/libevm/rlp"
+	"go.uber.org/zap"
+
+	"github.com/ava-labs/strevm/adaptor"
+
+	// Imported to allow IDE resolution of comments like [types.Block]. The
+	// package is imported in other files so this is a no-op beyond devex.
+	_ "github.com/ava-labs/libevm/core/types"
+)
+
+var _ adaptor.BlockProperties = (*Block)(nil)
+
+// ID returns [types.Block.Hash] from the embedded [types.Block].
+func (b *Block) ID() ids.ID {
+	return ids.ID(b.Hash())
+}
+
+// Parent returns [types.Block.ParentHash] from the embedded [types.Block].
+func (b *Block) Parent() ids.ID {
+	return ids.ID(b.ParentHash())
+}
+
+// Bytes returns the RLP encoding of the embedded [types.Block]. If encoding
+// returns an error, it is logged at the WARNING level and a nil slice is
+// returned.
+func (b *Block) Bytes() []byte {
+	buf, err := rlp.EncodeToBytes(b)
+	if err != nil {
+		b.log.Warn("RLP encoding error", zap.Error(err))
+		return nil
+	}
+	return buf
+}
```

Comment:
nit: Feels like the log should probably be an `Error`. Is this case ever expected?

Reply:
Done. Completely agree and have no idea why I used a lower level (this particular block of code is some of the earliest in the SAE implementation).